File: src/main/java/com/chrisdempewolf/fields/board/BoardFields.java
package com.chrisdempewolf.fields.board;

import com.chrisdempewolf.fields.BaseFields;

public class BoardFields extends BaseFields {

    @Override
    public BoardFields setAll() {
        return this.setCounts().setCreatedAt().setDescription().setId().setImage().setName().setUrl();
    }

    public BoardFields setId() {
        this.fields.add("id");
        return this;
    }

    public BoardFields setName() {
        this.fields.add("name");
        return this;
    }

    public BoardFields setUrl() {
        this.fields.add("url");
        return this;
    }

    public BoardFields setDescription() {
        this.fields.add("description");
        return this;
    }

    public BoardFields setCreatedAt() {
        this.fields.add("created_at");
        return this;
    }

    public BoardFields setCounts() {
        this.fields.add("counts");
        return this;
    }

    public BoardFields setImage() {
        this.fields.add("image");
        return this;
    }
}

File: src/main/java/com/chrisdempewolf/pinterest/fields/board/BoardFields.kt
package com.chrisdempewolf.pinterest.fields.board

import com.chrisdempewolf.pinterest.fields.BaseFields
import com.chrisdempewolf.pinterest.fields.pin.PinFields

class BoardFields : BaseFields() {

    override fun withAll(): BoardFields {
        return withCounts()
                .withCreatedAt()
                .withDescription()
                .withID()
                .withImage()
                .withName()
                .withURL()
    }

    fun withID(): BoardFields {
        fields = fields.plus("id")
        return this
    }

    fun withName(): BoardFields {
        fields = fields.plus("name")
        return this
    }

    fun withURL(): BoardFields {
        fields = fields.plus("url")
        return this
    }

    fun withDescription(): BoardFields {
        fields = fields.plus("description")
        return this
    }

    fun withCreatedAt(): BoardFields {
        fields = fields.plus("created_at")
        return this
    }

    fun withCounts(): BoardFields {
        fields = fields.plus("counts")
        return this
    }

    fun withImage(): BoardFields {
        fields = fields.plus("image")
        return this
    }

    fun with(field: String): BoardFields {
        fields = fields.plus(field)
        return this
    }
}
